src/shredder/pool.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Anza Technology, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Pool of shredder instances.
+//!
+//! # Examples
+//!
+//! Obtain a shredder from the pool, use it, and let it return on drop.
+//!
+//! ```rust
+//! use alpenglow::shredder::{RegularShredder, ShredderPool, Shredder};
+//!
+//! fn use_shredder<S: Shredder>(shredder: &mut S) {
+//!     // ...
+//! }
+//!
+//! let shredder_pool = ShredderPool::<RegularShredder>::with_size(1);
+//! {
+//!     let mut shredder = shredder_pool.take().unwrap();
+//!     use_shredder(&mut (*shredder));
+//!     // shredder is automatically returned to pool when dropped
+//! }
+//! ```
+
+use std::ops::{Deref, DerefMut};
+use std::sync::{Arc, Mutex};
+
+use super::Shredder;
+
+/// A pool of shredders of the same type.
+pub struct ShredderPool<S: Shredder> {
+    shredders: Arc<Mutex<Vec<S>>>,
+}
+
+impl<S: Shredder> ShredderPool<S> {
+    /// Creates a new pool with the provided shredders.
+    pub fn new(shredders: Vec<S>) -> Self {
+        Self {
+            shredders: Arc::new(Mutex::new(shredders)),
+        }
+    }
+
+    /// Takes a shredder from the pool.
+    ///
+    /// The shredder is automatically returned to the pool when dropped.
+    ///
+    /// Returns [`None`] iff the pool is empty.
+    pub fn take(&self) -> Option<ShredderGuard<S>> {
```

Comment:
`take` makes it sound like we are removing the shredder from the pool.  I initially thought that we were forgetting to return it to the pool in the code above.  I wonder if we should pick a different name.  `borrow_mut` might work better?  Not ideal either as we are not calling this function with `&mut self` which generally `borrow_mut` implies.